src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private int currentCapacity;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+        currentCapacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void add(T value) {
-
+        checkCapacity();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("This index is out of bounds.");
+        }
+        checkCapacity();
+        System.arraycopy(values, index, values, index + 1, size - index);
+        size++;
+        values[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T removedItem = values[index];
+        System.arraycopy(values,index + 1, values, index, size - index);
```

Comment:
```suggestion
        System.arraycopy(values, index + 1, values, index, size - index - 1);
```
we need to move one element less because we are removing

Reply:
done